src/validation.cpp

```diff
@@ -1571,8 +1571,10 @@ int ApplyTxInUndo(Coin&& undo, CCoinsViewCache& view, const COutPoint& out)
 }
 
 /** Undo the effects of this block (with given index) on the UTXO set represented by coins.
- *  When UNCLEAN or FAILED is returned, view is left in an indeterminate state. */
-static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& state, const CBlockIndex* pindex, CCoinsViewCache& view)
+ *  In case pfClean is provided, operation will try to be tolerant about errors, and *pfClean
+ *  will be true if no problems were found. Otherwise, the return value will be FAILED in case
+ *  of problems. Note that in any case, coins may be modified. */
+static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& state, const CBlockIndex* pindex, CCoinsViewCache& view, bool* pfClean = NULL)
 {
```

Comment:
ConnectBlock changes had already been made.

Reply:
Can I just drop 10178?

Reply:
see https://github.com/dashpay/dash/pull/2974/commits/d9a4dea613ecff9fb2293dfb59adbae9833a5561

Reply:
I'm fine with this :+1: